### PR TITLE
Add language and multi user collaboration support to onlyoffice

### PR DESCRIPTION
**Pull request by vigonotion**
This PR adds translation support to the onlyoffice editor by extracting the user language from the `Accept-Lang` header.

It also adds support for multi-user collaborative editing by removing the user id from the key for the onlyoffice document. I understand why you added that in, but I can't think of a case where the key has a hash collision with another totally different file because you already add the HashStream to the key. And in most cases, the user id is also already part of the path, so I'm pretty sure this is okay and enables users to use collaborative editing.

### CODE REVIEW

**Comment by vigonotion** [comment]
This is not stale, waiting for review from @mickael-kerjean

File: server/plugin/plg_editor_onlyoffice/index.go
Comment (mickael-kerjean) [inline server/plugin/plg_editor_onlyoffice/index.go:188]: I'm not sure why did you remove the userId there? It's a safety mechanism

File: server/plugin/plg_editor_onlyoffice/index.go
Comment (mickael-kerjean) [inline server/plugin/plg_editor_onlyoffice/index.go:246]: This will potentially create runtime errors when lang isn't the expected size, eg: https://play.golang.org/p/9c5DiECSFjK

**Review by mickael-kerjean — CHANGES_REQUESTED** [review]

File: server/plugin/plg_editor_onlyoffice/index.go
Comment (vigonotion) [reply server/plugin/plg_editor_onlyoffice/index.go:188]: I wrote why I had to do that in the second paragraph of my first comment

**Comment by mickael-kerjean** [comment]
This PR has been staying idle for a while already. If the issues aren't fixed I will be closing in the next 2-3 months

**Comment by vigonotion** [comment]
Sorry, completely forgot about this one. What do you think about the removal of the user id? It is necessary because otherwise multiple users cannot work together.

I will fix the potential runtime error

**Comment by rinkp** [comment]
While some backends may return different files for different users (e.g. \\fileserver\home) with the same path, perhaps a setting per backend would be nice to indicate that files are the same for all users _if they can see a certain file_. 
I think adding this would be a useful feature

**Pull request closed without merging**
